main.rb

```diff
@@ -0,0 +1,38 @@
+def bubble_sort(arr)
+  i = 0
+  while i < arr.length
+    j = 1
+    while j < arr.length - i
+      if arr[j - 1] > arr[j]
+        temp = arr[j - 1]
+        arr[j - 1] = arr[j]
+        arr[j] = temp
+      end
+      j += 1
+    end
+    i += 1
+  end
+  puts '[' + arr.join(',') + ']'
+end
+
+def bubble_sort_by(arr)
```

Comment:
This must return 
```
  > bubble_sort_by(["hi","hello","hey"]) do |left,right|
  >   left.length - right.length
  > end
  => ["hi", "hey", "hello"]
```
instead of 
```
  > bubble_sort_by(["hi","hello","hey"]) do |left,right|
  >   left.length - right.length
  > end
  => ["hy","hello","hi"]
 
```